### ISSUE TITLE
CORSActionBuilder calls Accumulator.done twice

### ISSUE THREAD

**Issue by wsargent** [playframework/playframework#8313]
In `CORSActionBuilder`, the accumulator is called twice, once inside the match block and again to return it.  This is probably a bug.

```scala
trait CORSActionBuilder extends ActionBuilder[Request, AnyContent] with AbstractCORSPolicy {

  protected def mat: Materializer

  override protected val logger: Logger = Logger.apply(classOf[CORSActionBuilder])

  override def invokeBlock[A](request: Request[A], block: Request[A] => Future[Result]): Future[Result] = {
    val action = new EssentialAction {
      override def apply(req: RequestHeader): Accumulator[ByteString, Result] = {
        req match {
          case r: Request[A] => Accumulator.done(block(r))
          case _ => Accumulator.done(block(req.withBody(request.body)))
        }
        Accumulator.done(block(req.asInstanceOf[Request[A]]))
      }
    }

    filterRequest(action, request).run()(mat)
  }
}
```

https://github.com/playframework/playframework/blob/master/framework/src/play-filters-helpers/src/main/scala/play/filters/cors/CORSActionBuilder.scala#L33

**Comment by marcospereira** [comment]
Yes, looks like a leftover after some refactoring.

**Comment by marcospereira** [comment]
By the way, there is https://github.com/playframework/playframework/pull/8230 already.

**Comment by marcospereira** [comment]
Fixed by #8230.
